bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/flamingos2/Flamingos2.java

```diff
@@ -889,13 +914,26 @@ public double getSetupTime(ISPObservation obs) {
         return getSpectroscopySetupSec();
     }
 
+    // for ITC overheads
+    public double getSetupTime(Config[] conf) {
+        if (isImaging(conf)) return getImagingSetupSec(conf);
+        return getSpectroscopySetupSec();
+    }
+
     /**
      * Is the instrument in imaging mode.
      */
     private boolean isImaging() {
         return (_fpu == FPUnit.FPU_NONE) && (_disperser == Disperser.NONE);
     }
 
+    private boolean isImaging(Config[] conf) {
+        _fpu = (FPUnit) conf[0].getItemValue(FPUnit.KEY);
+        _disperser = (Disperser) conf[0].getItemValue(Disperser.KEY);
+
+        return (_fpu == FPUnit.FPU_NONE) && (_disperser == Disperser.NONE);
```

Comment:
Why does this method mutate the value of `_fpu` and  `_disperser`?  It seems like it should be a `static` method that takes a `Config` and returns a value.

```java
private static boolean isImagingConfig(FPUnit fpu, Disperser disperser) {
    return (fpu == FPUnit.FPU_NONE) && (disperser == Dispserser.NONE);
}

private boolean isImagingConfig(Config conf) {
    return isImagingConfig(
        (FPUnit) conf.getItemValue(FPUnit.KEY),
        (Disperser) conf.getItemValue(Disperser.KEY));
}

private boolean isImaging() {
   return isImagingConfig(_fpu, _disperser);
}
```

Reply:
Ok, I have changed this as suggested.